src/main/rendering/ImageWriter.java

```diff
@@ -26,13 +44,18 @@ public ImageWriter(String filename, Resolution resolution) {
 	 * @param colour The colour to be written to the pixel.
 	 */
 	public void setPixel(int row, int col, Colour colour) {
-		// TODO: implement
+		pixels[row * width + col] = colour.rgb();
 	}
 
 	/**
 	 * Write the image to disk.
 	 */
 	public void writeToFile() {
-		// TODO: implement
+		try {
+			ImageIO.write(image, "PNG", new File( PATH + filename + EXT));
```

Comment:
```suggestion
			ImageIO.write(image, "JPG", new File(filename));
```
As per my previous comment.
Also PNG supports transparent background, thus JPG is more efficient if we don't need that. Any reason why you changed to PNG?

Reply:
Again, instructions. Doesn't really matter to me.